server/routes/api/pins/schema.ts

```diff
@@ -24,6 +24,21 @@ export const PinsCreateSchema = BaseSchema.extend({
 
 export type PinsCreateReq = z.infer<typeof PinsCreateSchema>;
 
+export const PinsInfoSchema = BaseSchema.extend({
+  body: z.object({
+    /** Document to get the pin info for. */
+    documentId: z
```

Comment:
nbd but I added a helper for this `zodIdType0`

```suggestion
    documentId: zodIdType(),
```

Reply:
Ah nice!